src/select.js

```diff
@@ -244,8 +244,8 @@
       function updateGroups(items) {
         ctrl.groups = [];
         angular.forEach(items, function(item) {
-          var groupFn = $scope.$eval(groupByExp);
-          var groupName = angular.isFunction(groupFn) ? groupFn(item) : item[groupFn];
+          var groupFn = $parse($scope.$eval(groupByExp));
```

Comment:
This is incorrect usage of $parse. $parse is used such as
```js
$parse(groupByExp)($scope)
```
This is essentially equivalent to $scope.$eval as far as this use goes though.